app/org/sagebionetworks/bridge/dynamodb/DynamoUploadDao.java

```diff
@@ -98,6 +107,20 @@ public Upload getUpload(@Nonnull String uploadId) {
 
     /** {@inheritDoc} */
     @Override
+    public List<? extends Upload> getStudyUploads(StudyIdentifier studyId, DateTime startTime, DateTime endTime) {
+        RangeKeyCondition condition = new RangeKeyCondition("requestedOn").between(
+                startTime.getMillis(), endTime.getMillis());
+        
+        List<DynamoUpload2> keysToGet = studyIdRequestedOnIndex.queryKeys(DynamoUpload2.class, "studyId",
+                studyId.getIdentifier(), condition);
+
+        return keysToGet.stream().map(key -> {
+            return mapper.load(key);
+        }).collect(Collectors.toList());
+    }
```

Comment:
DynamoIndexHelper.query() instead of queryKeys() will save you the work of re-querying the table yourself.